### PR TITLE
Add lib script, rocky and almalinux support

### PR DESCRIPTION
**Pull request by Linux123123**
WIP, not sure how #280 PR will adapt  / be merged.

Currently `panel.sh` has been tested in Ubuntu 20.04 and is at least partially working.

I moved a lot of redundant logic from scripts to lib, so I may have forgotten about something. My goal is to unify as much as possible all the scripts to have the same functions available to minimize overlap.

- [x] `panel`
- [x] `wings`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,9 +54,10 @@ List of supported installation setups for panel and Wings (installations support
 |                  | 11      | :white_check_mark: |
 | CentOS           | 6       | :red_circle:       |
 |                  | 7       | :white_check_mark: |
-|                  | 8       | :white_check_mark: |
+|                  | 8       | :red_circle: \*\*  |
 
 _\* Ubuntu 16 and Debian 8 no longer supported since Pterodactyl does not actively support it._
+_\*\* Centos 8 is EOL._
```
Comment (vilhelmprytz) [inline README.md:60]:
```suggestion
_\*\* CentOS 8 is EOL._
```

File: lib.sh [hunk 2]
```diff
@@ -0,0 +1,321 @@
+#!/bin/bash
+
+set -e
+
+#############################################################################
+#                                                                           #
+# Project 'pterodactyl-installer'                                           #
+#                                                                           #
+# Copyright (C) 2018 - 2021, Vilhelm Prytz, <vilhelm@prytznet.se>           #
```
Comment (vilhelmprytz) [inline lib.sh:9]:
```suggestion
# Copyright (C) 2018 - 2022, Vilhelm Prytz, <vilhelm@prytznet.se>           #
```

**Comment by Linux123123** [comment]
@vilhelmprytz the `installers/panel.sh` and `ui/panel.sh` are working (at least on ubuntu jammy) and should be ready for review. I want your opinion before I fully start working on `wings` part as to know what is right and what is wrong.

_*Ignore the failed builds*_

**Review by vilhelmprytz — COMMENTED** [review]
Initial read-through: this looks very promising, nice work @Linux123123! It looks like you've done some work on Wings as well already, would it be suitable if I merged this into a new branch and continued your work on it? If so, would you be able to resolve the current conflicts with `master`?

**Comment by Linux123123** [comment]
> Initial read-through: this looks very promising, nice work @Linux123123! It looks like you've done some work on Wings as well already, would it be suitable if I merged this into a new branch and continued your work on it? If so, would you be able to resolve the current conflicts with `master`?

I think I have the merge conflicts solved locally already, but yes we can continue on a new branch.

**Comment by Linux123123** [comment]
**Now to check for the millions of bugs introduced in this PR...**

 I will mark it as ready for review as in vagrant on ubuntu jammy I can run both panel and wings scritps without any major issues.

**Comment by vilhelmprytz** [comment]
Thanks for this, I'll merge into a new branch (`major-refactor`) and continue work there.

**Pull request merged**